### PR TITLE
Add parsedouble mapper function

### PR DESCRIPTION
**Pull request by aurrelhebert**

### CODE REVIEW

File: warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
+package io.warp10.script.mapper;
+  
+  //
+//Copyright 2016  Cityzen Data
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java]: we're in 2017 now :-)

File: warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import java.text.ParseException;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+* Mapper which returns the absolute value of the value passed as parameter
+*/
+public class MapperParseDouble extends NamedWarpScriptFunction implements WarpScriptMapperFunction {
+
+private final NumberFormat format;
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java]: indent

File: warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  }
+  
+  public MapperParseDouble(String name, Object language) throws WarpScriptException {   
+    super(name);
+    
+    if (language instanceof String) {
+      Locale locale = Locale.forLanguageTag((String) language);
+      format = NumberFormat.getInstance(locale);
+    } else {
+      throw new WarpScriptException("Invalid value type for " + getName() + ", expects a String");
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java]: Change message to (getName() + " expects a language tag (a string) on top of the stack.")

File: warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    public Builder(String name) {
+      super(name);
+    }
+    
+    @Override
+    public Object apply(WarpScriptStack stack) throws WarpScriptException {
+      Object value = stack.pop();
+      
+      if (!(value instanceof String)) {
+        throw new WarpScriptException("Invalid parameter for " + getName());
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java]: Use the same message as suggested above

File: warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+   */
+  public static class Builder extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+    
+    public Builder(String name) {
+      super(name);
+    }
+    
+    @Override
+    public Object apply(WarpScriptStack stack) throws WarpScriptException {
+      Object value = stack.pop();
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java]: Rename 'value' into something more meaningful

File: warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    Object value = null;
+    long location = locations[0];
+    long elevation = elevations[0];
+    
+    if (null != values[0]) {
+      if (values[0] instanceof String) {
+        try {
+          if (null == this.format) {
+            value = Double.parseDouble((String) values[0]);
+          } else
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java]: use '} else {'

File: warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    }
+    
+    Object value = null;
+    long location = locations[0];
+    long elevation = elevations[0];
+    
+    if (null != values[0]) {
+      if (values[0] instanceof String) {
+        try {
+          if (null == this.format) {
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java]: If this case possible? Does 'value instanceof String' return true when value is null?

Comment (aurrelhebert) [inline warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java]: Not anymore

Comment (aurrelhebert) [inline warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java]: it's deleted

File: warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+  }
+  
+  public MapperParseDouble(String name, Object language) throws WarpScriptException {   
+    super(name);
+    
+    if (language instanceof String) {
+      Locale locale = Locale.forLanguageTag((String) language);
+      format = NumberFormat.getInstance(locale);
+    } else {
+      throw new WarpScriptException("Invalid value type for " + getName() + ", expects a language tag (a string) on top of the stack.");
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java]: Modify the message by replacing 'value' by 'parameter'

File: warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+    public Builder(String name) {
+      super(name);
+    }
+    
+    @Override
+    public Object apply(WarpScriptStack stack) throws WarpScriptException {
+      Object language = stack.pop();
+      
+      if (!(language instanceof String)) {
+        throw new WarpScriptException("Invalid value type for " + getName() + ", expects a language tag (a string) on top of the stack.");
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/mapper/MapperParseDouble.java]: ditto above

**Pull request merged**